gym/spaces/box.py

```diff
@@ -78,16 +78,19 @@ def __init__(
 
         # determine shape if it isn't provided directly
         if shape is not None:
-            shape = tuple(shape)
-        elif not np.isscalar(low):
-            shape = low.shape  # type: ignore
-        elif not np.isscalar(high):
-            shape = high.shape  # type: ignore
+            shape = tuple(int(dim) for dim in shape)
```

Comment:
Why add the explicit type casting? I think this would mainly turn exceptions into weird, 99% undesired behavior which quietly goes through. Currently if some mad lad tries to use a float in a shape, it will crash a little further down the line in `np.full` which I think would be a pretty understandable exception, so we don't need to change anything here imo

Reply:
Agreed, I have just added an assert that all dims are `np.integers` so the user should get a clean, more understandable error message now